### DISCUSSION TITLE
POLL NEW FEATURE: Add `CupertinoTheme` support?

### DISCUSSION THREAD

**Discussion by rydmike** [rydmike/flex_color_scheme#69] [Polls]
This request exists also as feature request in the repo issues here: https://github.com/rydmike/flex_color_scheme/issues/20

When using `MaterialApp` and `FlexColorScheme` to make `ThemeData`, if using Cupertino widgets in such a `MaterialApp` they already get themed in a way that that is mapped from the Material theme to the Cupertino theme that those widgets use. This is done automatically by `ThemeData`.

However, currently `FlexColorScheme` cannot create a `CupertinoThemeData` that could be used to theme a `CupertinoApp`.

It is possible to add a `toCupertinoTheme` method to `FlexColorScheme` that would return a `CupertinoThemeData` object based on all given `FlexColorScheme` properties. This theme would however be much more limited and not be able to deliver all the things one can do with `ThemeData`. This is because Cupertino theming capabilities are much more limited in Flutter, as well as in Apple's human interface guide, it has less features than e.g. Material 3.

Since a `CupertinoThemeData` only has these theme properties and no sub-themes, there is really not much one can do to it:

* barBackgroundColor
* brightness
* primaryColor
* primaryContrastingColor 
* scaffoldBackgroundColor 
* textTheme

Still if it feels like a worthwhile addition, it could be added. The above properties correspond directly to existing ones in `FlexColorScheme` and can easily be mapped to a `CupertinoThemeData` in addition to `ThemeData`.

If you vote yes, please discuss and explain what a `FlexColorScheme` `CupertinoTheme` should include, support and do?

The default color mappings from `ThemeData` and its active `ColorScheme` colors to above and correct colors in `CupertinoThemeData` is already included via `ThemeData`'s default mappings. Which to be honest, while they can be changed, they can't be improved. So there is typically no real reason to change them, and if there really is, it can already be accomplished with `copyWith` for those rare occasions.

**Comment by fotiDim** [comment]
Yes, please! I am using [PlatformApp](https://github.com/stryder-dev/flutter_platform_widgets/blob/master/README.md#platformapp) which is an abstraction over MaterialApp and CupertinoApp. Theming is an issue since for Cupertino instead of separate light and dark themes, we need to use `CupertinoDynamicColor` for each color individually and specify its light/dark value, if their [docs](https://github.com/stryder-dev/flutter_platform_widgets/blob/434af6d2141debcbc8b150ac8dd1eccfd6857063/docs/theming.md#cupertino) are accurate.
One could use `MaterialBasedCupertinoThemeData( materialTheme: theme)` to get a cupertino theme out of a material one but there is no easy way to set a separate light and dark themes which leads to more workarounds.
It would be great if we could get `CupertinoThemeData` with all the light/dark values preconfigured.

**Reply by rydmike** [reply]
Thanks @fotiDim for the insights and response. I do want to add support for `CupertinoApp` at some point going forward too. 

I had not considered the `PlatformApp` case in the package `https://pub.dev/packages/flutter_platform_widgets`, but it sounds interesting, I will use their example as well when I attack `CupertinoThemeData` to see what I need in FCS API to support it. In addition to a new `CupertinoApp` example, I could use a modified version of their example as a `PlatformApp` demo using `FCS` as well.

I have not used `flutter_platform_widgets` in any real project, but I played with it maybe 3 years ago when it was still very new. It would be interesting to take a look at it again.

**Comment by fredgrott** [comment]
I vote yes, as the general case I am looking at is using Flutter Platform Widgets along with doing platform builders for fluent(ms desktop) widgets. In which I would be setting Apple devices to the default CupertinoTheme without the added Material Theme stuff.

That being said, it is generally mappable right now given the updated stuff you did in FCS 6. I guess I will find out as this week I am digging in and implementing examples.

**Comment by ollerandres** [comment]
I can help with this. It would be really useful for the community as 100% of users voted for a YES.

**Reply by rydmike** [reply]
I have been wondering in the past what users would expect this feature to actually do? It is discussed a bit in this issue https://github.com/rydmike/flex_color_scheme/issues/20

I'm wondering because iOS Cupertino widgets in Flutter can basically not be themed at all (yet, they may get some capabilities later), but currently they are what they are. Apart from a few colors that can be assigned to the Cupertino theme.

When you use `MaterialApp` with its `ThemeData`, the colors from `ColorScheme` are already mapped to these Cupertino theme colors. There is currently nothing else that can be done. Sure you could do some of the color mappings from ColorScheme to the Cupertino limited and few theme colors available, a bit differently, but there is no practical reason to do so, as they are already correctly mapped by `ThemeData` (also with FlexColorScheme) and as good as they can be.

Regardless of this, I am curious about what users would expect "Cupertino" support to do currently?

I added this questionnaire because I have been eager to find out, so any input is welcome. As a first help step, I would love to hear what your ideas on the topic are 😄 


If/when more theming support arrives in Flutter's Cupertino widgets, I do plan to add and support them.

I am also considering a more general design of the Playground app that could include support for some visual theming of other popular 3rd party design systems and components too.
